packages/rest/src/parser.ts

```diff
@@ -103,13 +105,14 @@ function buildOperationArguments(
     const spec = paramSpec as ParameterObject;
     switch (spec.in) {
       case 'query':
-        paramArgs.push(request.query[spec.name]);
+        paramArgs.push(paramCoerce(request.query[spec.name], spec.schema));
```

Comment:
The bit `paramArgs.push(paramCoerce` is duplicated many times and it's getting rather long. Could you please clean up this code to avoid repetition please?

I think we should move the `switch (spec.in)` statement to a new function that will return the value from the request, and then keep `paramArgs.push(paramCoerce)` in a single place here.

```ts
const spec = paramSpec as ParameterObject;
const rawValue = getParamFromRequest(request, spec.in);
const value = paramCoerce(rawValue, spec.schema);
paramArgs.push(value);
```